Add docs: Deploy deployment into a specified set of target clusters (#1231)
karmada-io/karmada: docs/userguide/resource-propagating.md

## Deploy deployment into a specified set of target clusters

> `.spec.placement.clusterAffinity` field of PropagationPolicy represents scheduling restrictions to a certain set of clusters. If not set, any cluster can be scheduling candidate.

Reviewer: ```suggestion
`.spec.placement.clusterAffinity` field of PropagationPolicy represents scheduling restrictions on a certain set of clusters, without which any cluster can be scheduling candidates.
```

It has four fields to set:
- LabelSelector
- FieldSelector
- ClusterNames
- ExcludeClusters

### LabelSelector

> LabelSelector is a filter to select member clusters by labels, it uses `*metav1.LabelSelector` type. If non-nil and non-empty, only the clusters match this filter will be selected.

Reviewer: ```suggestion
LabelSelector is a filter to select member clusters by labels. It uses `*metav1.LabelSelector` type. If it is non-nil and non-empty, only the clusters match this filter will be selected.
```

PropagationPolicy can be configured as follows:

```yaml
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: test-propagation
spec:
  ...
  placement:
    clusterAffinity:
      labelSelector:
        matchLabels:
          location: us
    ...
```

PropagationPolicy can also be configured as follows:

```yaml
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: test-propagation
spec:
  ...
  placement:
    clusterAffinity:
      labelSelector:
        matchExpressions:
        - key: location
          operator: In
          values:
          - us
    ...
```

For a description of `matchLabels` and `matchExpressions`, you can refer to [Resources that support set-based requirements](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements).

### FieldSelector

> FieldSelector is a filter to select member clusters by fields. If non-nil and non-empty, only the clusters match this filter will be selected.

Reviewer: ```suggestion
FieldSelector is a filter to select member clusters by fields. If it is non-nil and non-empty, only the clusters match this filter will be selected.
```

PropagationPolicy can be configured as follows:

```yaml
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: nginx-propagation
spec:
  ...
  placement:
    clusterAffinity:
      fieldSelector:
        matchExpressions:
        - key: provider
          operator: In
          values:
          - huaweicloud
        - key: region
          operator: NotIn
          values:
          - cn-south-1
    ...
```

If multiple `matchExpressions` are specified in the `fieldSelector`, the cluster must match all `matchExpressions`.

> The `key` value in `matchExpressions` field now support three values: `provider`, `region`, `zone`, which correspond to the `.spec.provider`, `.spec.region`, `.spec.zone` fields of the Cluster object respectively.

Reviewer: ```suggestion
The `key` in `matchExpressions` now supports `provider`, `region`, and `zone`, which correspond to the `.spec.provider`, `.spec.region`, and `.spec.zone` fields of the Cluster object, respectively.
```

> The `operator` value in `matchExpressions` field now support two operator: `In` and `NotIn`.

Reviewer: ```suggestion
The `operator` in `matchExpressions` now supports `In` and `NotIn`.
```

### ClusterNames

> User can set the ClusterNames field to specify the name of the selected cluster list.

Reviewer: ```suggestion
Users can set the `ClusterNames` field to specify the selected clusters.
```

PropagationPolicy can be configured as follows:

```yaml
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: nginx-propagation
spec:
  ...
  placement:
    clusterAffinity:
      clusterNames:
        - member1
        - member2
    ...
```

### ExcludeClusters

> User can set the ExcludeClusters fields to specify the cluster name list, indicating that these clusters are ignored.

Reviewer: ```suggestion
User can set the `ExcludeClusters` fields to specify the clusters to be ignored.
```
